python/ray/data/_internal/stats.py

```diff
@@ -407,8 +407,9 @@ def clear_execution_metrics(self, dataset_tag: str, operator_tags: List[str]):
 
     def clear_iteration_metrics(self, dataset_tag: str):
         tags = self._create_tags(dataset_tag)
-        self.iter_total_blocked_s.set(0, tags)
-        self.iter_user_s.set(0, tags)
+        # NOTE(rickyx): We should not be clearing the iter_total_blocked_s and
+        # iter_user_s metrics because they are technically counters we tracked, and
+        # should not be reset by each iteration.
```

Comment:
@rickyyx could you help me understand why this value keeps going up across iterations? Is it because we reset the value here, but not in some other place?

Reply:
During my investigation for https://github.com/ray-project/ray/issues/44635, where the `Rows Outputted` value is seen as zero, I found that disabling `clear_execution_metrics()` [here](https://github.com/scottjlee/ray/blob/master/python/ray/data/_internal/execution/streaming_executor.py#L193-L195) is the fix. 

I was thinking it would make sense to just completely remove `clear_execution_metrics()` and `clear_iteration_metrics()` calls currently being called after Dataset execution/iteration completes -- @rickyyx do you agree? For more context, I believe the reason why we had this in the first place is to do this hacky "reset" of metrics, to prevent values from persisting at the last value. But I think this is no longer the behavior we want, since we also now show rates on the Grafana dashboard by default -- so we can simply remove the metrics reset.

Reply:
+1 to just removing it 

Why also not converting these to counters as well?

Reply:
Sure - i think removing it makes sense. 

> Why also not converting these to counters as well?

Yeah, I think this was an alternative, but if I remember correctly, we are less prone to changing actual timeseries definition since there might be customers depending on this? (or what's the policy here for backward compatibility on the metrics)?

Reply:
> could you help me understand why this value keeps going up across iterations

I think it's because we reuse the stats field. Let me see if that could be fixed.